contracts/serverless-v2/Jobs.sol

```diff
@@ -0,0 +1,456 @@
+
+        // on reward distribution, 1st output executor node gets max reward
+        // reward ratio - 4:3:2
+        _transferRewardPayout(_jobId, outputCount, enclaveAddress);
+
+        // TODO: add callback gas
+        if (outputCount == 1) {
+            address jobOwner = jobs[_jobId].jobOwner;
+            (bool success, ) = jobOwner.call(
+                abi.encodeWithSignature(
+                    "oysterResultCall(uint256,bytes,uint8,uint256)",
+                    _jobId,
+                    _output,
+                    _errorCode,
+                    _totalTime
+                )
+            );
+            emit JobSucceeded(_jobId, success);
+        }
+        emit JobResponded(_jobId, _output, _totalTime, _errorCode, outputCount);
+    }
+
+    function _verifyOutputSign(
+        bytes memory _signature,
+        uint256 _jobId,
+        bytes memory _output,
+        uint256 _totalTime,
+        uint8 _errorCode,
+        uint256 _signTimestamp
+    ) internal view returns (address) {
+        if (block.timestamp > _signTimestamp + SIGN_MAX_AGE) revert JobsSignatureTooOld();
+
+        bytes32 hashStruct = keccak256(
+            abi.encode(SUBMIT_OUTPUT_TYPEHASH, _jobId, keccak256(_output), _totalTime, _errorCode, _signTimestamp)
+        );
+        bytes32 digest = keccak256(abi.encodePacked("\x19\x01", DOMAIN_SEPARATOR, hashStruct));
+        address signer = digest.recover(_signature);
+
+        EXECUTORS.allowOnlyVerified(signer);
+        return signer;
+    }
+
+    function _transferRewardPayout(uint256 _jobId, uint256 _outputCount, address _enclaveAddress) internal {
+        address owner = EXECUTORS.getOwner(_enclaveAddress);
+        uint256 executionTime = jobs[_jobId].executionTime;
+        address jobOwner = jobs[_jobId].jobOwner;
+        uint256 deadline = jobs[_jobId].deadline;
+        uint256 executorsFee = executionTime * EXECUTOR_FEE_PER_MS;
+        // for first output
+        if (_outputCount == 1) {
+            // transfer payout to executor
+            USDC_TOKEN.safeTransfer(owner, (executorsFee * 4) / 9);
+            // transfer payout to payment pool
+            USDC_TOKEN.safeTransfer(USDC_PAYMENT_POOL, executionTime * STAKING_REWARD_PER_MS);
+            // transfer to job owner
+            USDC_TOKEN.safeTransfer(
+                jobOwner,
+                (deadline - executionTime) * (EXECUTOR_FEE_PER_MS + STAKING_REWARD_PER_MS)
+            );
+        }
+        // for second output
+        else if (_outputCount == 2) {
+            // transfer payout to executor
+            USDC_TOKEN.safeTransfer(owner, executorsFee / 3);
+        }
+        // for 3rd output
+        else {
+            // transfer payout to executor
+            USDC_TOKEN.safeTransfer(owner, executorsFee - (executorsFee * 7) / 9);
+            // cleanup job data after 3rd output submitted
+            _cleanJobData(_jobId);
+        }
```

Comment:
does this always add up even in cases of truncating? basically `(executorsFee * 4) / 9` and `executorsFee / 3` will both truncate off, so essentially truncated twice, while `(executorsFee * 7) / 9` will truncate once

Reply:
yep, this won't add up always. Fixed it.